### PR TITLE
166 bidding pallet execute bidding workflow periodically

### PR DESCRIPTION
**Pull request by letodunc**
Added periodic call to the bidding pallet
- Added `BiddingExecutionPeriod` in Bidding config pallet
- Added hook to call each `BiddingExecutionPeriod` the bidding workflow

### CODE REVIEW

File: pallets/bidding/src/lib.rs [hunk 1]
```diff
@@ -92,6 +94,14 @@ pub mod pallet {
 		NoNewHousesFound,
```
Comment (ilhanu) [inline pallets/bidding/src/lib.rs:94]: This doesn't belong here. If there are no new houses it should not emit an Error, but just an event is fine

File: pallets/bidding/src/lib.rs [hunk 2]
```diff
@@ -92,6 +94,14 @@ pub mod pallet {
...
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Weight: see `begin_block`
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			Self::begin_block(n)
+		}
+	}
+
```
Comment (ilhanu) [inline pallets/bidding/src/lib.rs:104]: How does this exactly work? 
The hook checks every block if it's time to starting the Bidding ?

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -1053,6 +1053,7 @@ parameter_types! {
 	pub const MaxTriesAseemblingInvestor: u64 = 3;
 	pub const MaximumSharePerInvestor: u64 = 20;
 	pub const MinimumSharePerInvestor: u64 = 10;
+	pub const BiddingExecutionPeriod: u32 = EPOCH_DURATION_IN_BLOCKS;
```
Comment (ilhanu) [inline runtime/src/lib.rs:1056]:
```suggestion
	pub const NewAssetScanPeriod: u32 = EPOCH_DURATION_IN_BLOCKS;
```

File: pallets/bidding/src/lib.rs [hunk 4]
```diff
@@ -59,6 +59,8 @@ pub mod pallet {
 		type MaxTriesAseemblingInvestor: Get<u64>;
 		type MaximumSharePerInvestor: Get<u64>;
 		type MinimumSharePerInvestor: Get<u64>;
+		#[pallet::constant]
+		type BiddingExecutionPeriod: Get<Self::BlockNumber>;
```
Comment (ilhanu) [inline pallets/bidding/src/lib.rs:63]:
```suggestion
		type NewAssetScanPeriod: Get<Self::BlockNumber>;
```

File: runtime/src/lib.rs [hunk 5]
```diff
@@ -1064,6 +1065,7 @@ impl pallet_bidding::Config for Runtime {
 	type MaxTriesAseemblingInvestor = MaxTriesAseemblingInvestor;
 	type MaximumSharePerInvestor = MaximumSharePerInvestor;
 	type MinimumSharePerInvestor = MinimumSharePerInvestor;
+	type BiddingExecutionPeriod = BiddingExecutionPeriod;
```
Comment (ilhanu) [inline runtime/src/lib.rs:1068]:
```suggestion
	type NewAssetScanPeriod = NewAssetScanPeriod;
```

**Review by ilhanu — CHANGES_REQUESTED** [review]
- When bidding-pallet begins, we want to have these periodic events to be published. A good event name would be `NewAssetScanPeriod`
For the rest looking good and happy to test it further once merged into main

File: pallets/bidding/src/lib.rs (hunk 2 above)
Comment (letodunc) [reply pallets/bidding/src/lib.rs:104]: It checks every block if an epoch has finished to launch the process.

**Comment by ilhanu** [comment]
@letodunc can we test that the hook works as intended ? 

- New session should start the hook
- The hook should make the bidding pallet scan

**Review by ndkazu — APPROVED** [review]

**Comment by letodunc** [comment]
> Please before accepting we need the test for the hook transition

I propose to do the test in a second step. For simpliciy, let's merge the #168 and this PR in main as there are changes in #168 impacting #166. The new PR will contains test on periodic call.

**Pull request merged**